backends/gstreamer/src/player.rs

```diff
@@ -290,6 +292,30 @@ impl PlayerInner {
         }
         Ok(result)
     }
+
+    fn set_stream(&mut self, mut stream: Box<MediaStream>) -> Result<(), PlayerError> {
+        debug_assert!(self.stream_type == StreamType::Stream);
+        if let Some(ref source) = self.source {
+            if let PlayerSource::Stream(source) = source {
+                if let Some(mut stream) = stream.as_mut_any().downcast_mut::<GStreamerMediaStream>()
+                {
+                    let playbin = self
+                        .player
+                        .get_pipeline()
+                        .dynamic_cast::<gst::Pipeline>()
+                        .unwrap();
+                    let clock = gst::SystemClock::obtain();
+                    playbin.set_base_time(clock.get_time());
```

Comment:
This base time might not be exactly the same as the one set on the mediastream pipeline. You'd need to share the value, somehow.

Reply:
I think we'll probably try to set basetime zero everywhere to be consistent?